queries.js

```diff
@@ -0,0 +1,179 @@
+const dayjs = require('dayjs');
+
+module.exports = (self, query) => {
+  return {
+    builders: {
+      future: {
+        def: null,
+        finalize() {
+          const future = query.get('future');
```

Comment:
To avoid granting inappropriate access through public REST APIs or any other side door we still need to always act as if `future === false` when `self.apos.permission.can(req, 'edit', self.name, 'draft')` is falsy. `itemIsVisible` checks the same thing in the admin bar module to decide if an admin bar item should be shown.

Reply:
Something like that?

```js
          const req = self.apos.task.getReq();
          if (!self.apos.permission.can(req, 'edit', self.name, 'draft')) {
            future = false;
          }
          ```